omod/src/test/webapp/resources/scripts/navigatorHandlers.js

```diff
@@ -712,21 +713,27 @@ describe("Tests for simple form navigation handlers", function() {
                 expect(event.stopPropagation).toHaveBeenCalled();
                 expect(field.select).toHaveBeenCalled();
             });
-            it("should not switch selection to question ahead if question in between is not valid", function() {
+            it("should not switch selection to question ahead but should switch to invalid question in between", function() {
                 spyOn(event, 'stopPropagation');
                 spyOn(firstQuestion, 'isValid').and.returnValue(true);
                 spyOn(firstQuestion, 'onExit').and.returnValue(true);
+                spyOn(firstQuestion, 'toggleSelection');
                 spyOn(secondQuestion, 'isValid').and.returnValue(false);
+                spyOn(secondQuestion, 'toggleSelection');
+                spyOn(secondQuestion, 'firstInvalidField');
                 spyOn(field, 'select');
                 firstQuestion.isSelected=true;
                 field.isSelected = true;
                 firstQuestion.fields = [field];
+                spyOn(secondField, 'select');
+                spyOn(secondField, 'toggleSelection');
+                spyOn(secondField, 'isValid').and.returnValue(false);
+                secondQuestion.fields = [ secondField ];
 
                 clickedQuestionHandler(questions, thirdQuestion, event);
 
-                expect(firstQuestion.onExit).not.toHaveBeenCalled();
+                expect(firstQuestion.onExit).toHaveBeenCalled();
                 expect(event.stopPropagation).toHaveBeenCalled();
```

Comment:
Shouldn't the following new test cases be added?

```
expect(secondQuestion.toggleSelection).toHaveBeenCalled();
 expect(secondField.select).toHaveBeenCalled()
```

(Or at least the second of the two above?)

Reply:
Thanks @mogoodrich ! I have added the following two validation statements:
`expect(secondQuestion.toggleSelection).toHaveBeenCalled();
                expect(secondField.toggleSelection).toHaveBeenCalled();`

If you look at the handler code, the field.toggleSelection() function is called:
https://github.com/openmrs/openmrs-module-uicommons/pull/107/files#diff-843c954a225b484145f9fd50f0ef831655d7001490c56a23c9c1053f4c4ad1cbR300